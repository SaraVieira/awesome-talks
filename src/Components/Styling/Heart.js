import styled from 'styled-components'
import is from 'styled-is'
import remcalc from 'remcalc'

export default styled.div`
    position: absolute;
    z-index: 10;
    right: ${remcalc(10)};
    top: ${remcalc(10)};

    ${is('watched')`
      top: ${remcalc(50)};
  `};

    input[type='checkbox'] {
        clear: both;
        display: none;
    }

    input[type='checkbox'] {
        display: none;
    }

    input[type='checkbox'] + label {
        z-index: 100;
        overflow: hidden;
        text-align: center;
        cursor: pointer;
    }

    input[type='checkbox'] + label:before {
        content: '';
        z-index: -1;
        position: absolute;
        background: rgb(214, 214, 214);
        width: 100%;
        height: 100%;
        border-radius: 50%;
        top: 0;
        left: 0;
        transform: scale(0);
    }

    input[type='checkbox'] + label:after {
        content: '';
        z-index: -1;
        position: absolute;
        background: white;
        width: 100%;
        height: 100%;
        border-radius: 50%;
        top: 0;
        left: 0;
        transform: scale(0);
    }

    input[type='checkbox']:checked + label svg {
        transition: all 300ms ease-in-out;
        fill: ${props => props.theme.red};
        transform: scale(1.3);
        background: none;
    }

    input[type='checkbox']:checked + label svg g {
        fill: ${props => props.theme.green};
        background: none;
    }

    input[type='checkbox']:checked + label:after {
        animation: like-a 0.3s 0.2s both;
    }

    input[type='checkbox']:checked + label:before {
        animation: like-a 0.5s 0s both;
    }

    label svg,
    label g {
        transition: all 300ms ease-in-out;
        display: inline-flex;
        vertical-align: middle;
        width: ${remcalc(20)};
        height: ${remcalc(20)};
        fill: rgba(255, 255, 255, 0.95);
        border-radius: 50%;
        background: rgba(39, 39, 39, 0.35);
        padding: 6px;
    }
`
