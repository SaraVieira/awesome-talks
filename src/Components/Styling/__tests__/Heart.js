import React from 'react'
import Shallow from '../../../Tests/shallow'
import Heart from '../Heart'

test('Renders Heart', () => {
    const Wrapper = Shallow(<Heart />)

    // expect(Wrapper.find('svg')).toHaveStyleRule('fill', 'rgb(214, 214, 214)')
    expect(Wrapper).toMatchSnapshot()
})
