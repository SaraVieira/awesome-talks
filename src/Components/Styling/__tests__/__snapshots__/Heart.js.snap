// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Heart 1`] = `
.c0 {
  position: absolute;
  z-index: 10;
  right: 0.625rem;
  top: 0.625rem;
}

.c0 input[type='checkbox'] {
  clear: both;
  display: none;
}

.c0 input[type='checkbox'] {
  display: none;
}

.c0 input[type='checkbox'] + label {
  z-index: 100;
  overflow: hidden;
  text-align: center;
  cursor: pointer;
}

.c0 input[type='checkbox'] + label:before {
  content: '';
  z-index: -1;
  position: absolute;
  background: rgb(214,214,214);
  width: 100%;
  height: 100%;
  border-radius: 50%;
  top: 0;
  left: 0;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
}

.c0 input[type='checkbox'] + label:after {
  content: '';
  z-index: -1;
  position: absolute;
  background: white;
  width: 100%;
  height: 100%;
  border-radius: 50%;
  top: 0;
  left: 0;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
}

.c0 input[type='checkbox']:checked + label svg {
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  -webkit-transform: scale(1.3);
  -ms-transform: scale(1.3);
  transform: scale(1.3);
  background: none;
}

.c0 input[type='checkbox']:checked + label svg g {
  background: none;
}

.c0 input[type='checkbox']:checked + label:after {
  -webkit-animation: like-a 0.3s 0.2s both;
  animation: like-a 0.3s 0.2s both;
}

.c0 input[type='checkbox']:checked + label:before {
  -webkit-animation: like-a 0.5s 0s both;
  animation: like-a 0.5s 0s both;
}

.c0 label svg,
.c0 label g {
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  width: 1.25rem;
  height: 1.25rem;
  fill: rgba(255,255,255,0.95);
  border-radius: 50%;
  background: rgba(39,39,39,0.35);
  padding: 6px;
}

<div
  className="c0"
/>
`;
